treeherder/client/thclient/client.py

```diff
@@ -674,7 +678,10 @@ def __init__(
     def _get_uri(self, project, endpoint, data=None, oauth_key=None,
                  oauth_secret=None, method='GET'):
 
-        uri = '{0}://{1}/api/project/{2}/{3}/'.format(
+        if method == 'POST':
+            endpoint += '/'
```

Comment:
It's probably a good idea to add a `append_slash=True` parameter to _get_uri and let the caller to decide whether to add a trailing slash or not. An alternative could be to have it as a parameter of the `TreeherderClient` constructor, if you prefer a more centralized place.